lib/src/data/data.rs

```diff
@@ -109,6 +110,14 @@ impl Data {
         fn concrete_stream(stream: &&mut NetworkStream) -> Option<NetStream> {
             stream.downcast_ref::<HttpStream>()
                 .map(|s| NetStream::Http(s.clone()))
+                .or_else(|| {
```

Comment:
This needs to happen whether TLS is or isn't enabled. If you move all of this functionality to a common `unix` module and expose (mostly) no-op shims on non-Unix platforms, this becomes:

```rust
stream.downcast_ref::<HttpStream>()
    .map(|s| NetStream::Http(s.clone()))
    .or_else(try_unix_socket_stream_downcast)
```